grafana-plugin/src/plugin/PluginSetup/index.tsx

```diff
@@ -34,7 +35,10 @@ const PluginSetupWrapper: FC<PluginSetupWrapperProps> = ({ text, children }) =>
 
 const PluginSetup: FC<PluginSetupProps> = observer(({ InitializedComponent, ...props }) => {
   const store = useStore();
-  const setupPlugin = useCallback(() => store.setupPlugin(props.meta), [props.meta]);
+  const setupPlugin = useCallback(
+    () => PluginState.getGrafanaPluginSettings().then((meta) => store.setupPlugin(meta)),
```

Comment:
for my own curiosity, how're `props.meta` and the results from `GET /api/plugins/grafana-oncall-app/settings` different here?

Reply:
This handles the case mentioned in the comment for `triggerPluginReset` where configuration has been removed but the page has not been reloaded (Still need to update that comment).  Could also help with some edge cases where the provisioned configuration has changed and reload has not happened yet.

Reply:
So I ended up changing to reload the page when configuration is removed since it was confusing to create a second `meta` object and use it instead of the one passed in.  Was there a reason we stopped reloading the page when configuration was removed? I thought we used to do that.

Reply:
this comment.. doh 🤦 
```tsx
/**
   * NOTE: there is a possible edge case when resetting the plugin, that would lead to an error message being shown
   * (which could be fixed by just reloading the page)
   * This would happen if the user removes the plugin configuration, leaves the page, then comes back to the plugin
   * configuration.
   *
   * This is because the props being passed into this component wouldn't reflect the actual plugin
   * provisioning state. The props would still have onCallApiUrl set in the plugin jsonData, so when we make the API
   * call to check the plugin state w/ OnCall API the plugin-proxy would return a 502 Bad Gateway because the actual
   * provisioned plugin doesn't know about the onCallApiUrl.
   *
   * This could be fixed by instead of passing in the plugin provisioning information as props always fetching it
   * when this component renders (via a useEffect). We probably don't need to worry about this because it should happen
   * very rarely, if ever
   */
```

I _believe_ it was removed when the new navigation came into effect, and there wasn't a need to reload to have the OnCall icon show up on the left hand nav